networkx/relabel.py

```diff
@@ -139,8 +140,8 @@ def _relabel_inplace(G, mapping):
                 "resolve the mapping. Use copy=True."
             ) from err
     else:
-        # non-overlapping label sets
-        nodes = old_labels
+        # non-overlapping label sets, sort them in the order of G nodes
+        nodes = list(n for n in G if n in mapping.keys())
```

Comment:
```suggestion
        nodes = [n for n in G if n in mapping]
```
I think this will be faster and easier to read.